src/main/java/hudson/plugins/s3/utils/S3Utils.java

```diff
@@ -0,0 +1,64 @@
+package hudson.plugins.s3.utils;
+
+import hudson.ProxyConfiguration;
+import hudson.util.Secret;
+
+import java.util.regex.Pattern;
+
+import jenkins.model.Jenkins;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.internal.StaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3Client;
+
+public class S3Utils {
+
+    public static AmazonS3Client createClient(String accessKey, Secret secretKey, boolean useRole, String stsRoleArn) {
+        AWSCredentialsProvider credentialsProvider;
+        if (useRole) {
+            credentialsProvider = new DefaultAWSCredentialsProviderChain();
```

Comment:
Thank you Cyrille for reviewing this.

I understand your comment, but IMHO, if the plugin allows anonymous access, it should be explicit. In other words, it should have an "anonymous" checkbox or radio in the configuration form. And it this should be tested.

Anyway, I don't think anonymous upload is a real need. I mean who gives anonymous upload access to their buckets? But I could be wrong. ;)

Cheers,
Eric.